### DISCUSSION TITLE
Does blink.cmp support `nofile` buftypes?

### DISCUSSION THREAD

**Discussion by olimorris** [saghen/blink.cmp#439] [Q&A] [answered]
I'm looking to add a source for my [plugin](https://github.com/olimorris/codecompanion.nvim), to blink.

For context, the buffer I would like blink to work in is a custom filetype but with a buftype of `nofile`. So with my custom source:

```lua
--codecompanion.integrations.blink
local M = {}

function M.new()
  return setmetatable({}, { __index = M })
end

function M:get_trigger_characters()
  return { "/" }
end

function M:enabled(context)
  if context then
    return vim.bo[context.bufnr].filetype == "codecompanion"
  end
  return true -- force this to be enabled for the purposes of this demo
end

function M:get_completions(context, callback)
  print("ENABLED")
end

return M
```

and my custom config:

```lua
-- ...
sources = {
  compat = {},
  completion = {
    enabled_providers = { "lsp", "path", "snippets", "buffer", "codecompanion" },
  },
  providers = {
    codecompanion = {
      name = "CodeCompanion",
      module = "codecompanion.integrations.blink",
      enabled = true,
    },
  },
},
-- ...
```

I will only ever see the printed message `enabled`, if I do `:set buftype=` and then hit the trigger character in the buffer.

Is this by design? Is there a way I could around this?

**Comment by saghen** [comment, marked as answer]
It automatically disables itself on `buftype ~= ''` but that definitely seems like something we should change. While you're waiting, you can change `is_blocked_buffer()` in `lib/utils.lua` to not check the `buftype`

**Reply by olimorris** [reply]
Awesome. Shall I transfer this to an issue?

**Reply by saghen** [reply]
That'd be great thanks!
